packages/embarkjs/src/blockchain.js

```diff
@@ -309,40 +310,47 @@ let Contract = function(options) {
     }
   });
 
+  // Assign helpers too
+  for(const method of ["deploy", "new", "at", "send", "deployed"]) {
+    // Make sure we don't override original methods here.
+    if (originalMethods.indexOf(method) >= 0) {
+      console.log(method + " is a reserved word and will not be aliased as a helper");
+      continue;
+    }
+
+    ContractClass[method] = Contract.prototype[method].bind(this);
+  }
```

Comment:
Considering that those methods have so far been defined on `Contract.prototype`, and the fact that the `Contract` constructor actually returns a different type than what the constructor itself is, those prototype methods are not accessible in the first place.

e.g.

```
const c = new EmbarkJS.Blockchain.Contract(...) // c is not actually of type `Contract`
```

Knowing that, I'd assume that these methods aren't used in the wild at the moment. I'm not even sure why we're having all this data (like abi etc) attached to `Contract`, if in reality we're given the caller a `ContractClass`.

With that in mind, I'd say there's actually no point in keeping those prototype methods around, as `Contract` acts more like a factory. We should probably move those methods to whatever is returnd from `blockchainConnector.newContract()`

Reply:
You're totally right. Should we leave that to another PR, where we discuss what work the connector should do and what work we should do?

Reply:
I think we can leave here... There's agreement in the team to overhaul the whole EmbarkJS side of things at some point anyways.